### PR TITLE
doc(AM-4043): Open Source Documentation

### PR DESCRIPTION
**Pull request by narmidm**
Signed-off-by: MD IMRAN <imranaec@outlook.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+* Contribute to the kubeslice codebase
+* Report and triage bugs.
+* Answer questions on KubeSlice discussions forums
+* Write technical documentation and blog posts, for users and contributors.
+* Filing issues
+* Provide feedback
+* Submit PRs for small bug fixes
+
+## Contributor compliance with Developer Certificate Of Origin (DCO)
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:18]: Change it to - Contributor Compliance with Developer Certificate Of Origin

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+We require every contributor to certify that they are legally permitted to contribute to our project. A contributor expresses this by consciously signing their commits, and by this act expressing that they comply with the Developer Certificate Of Origin
+
+A signed commit is a commit where the commit message contains the following content:
+
+`Signed-off-by: John Doe <jdoe@example.org>`
+
+This can be done by adding `--signoff` to your git command line
+
+## Code of conduct
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:28]: Change it to - Code of Conduct

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+4. **Push** your work back up to your fork
+5. Submit a **Pull request** so that we can review your changes
+
+NOTE: Be sure to merge the latest from "upstream" before making a pull request!
+
+## Filing Issues
+If you encounter a bug please file an issue in the Github repository. Please provide as much information as possible to help fix the issue. Please file the issue under the appropriate repo.
+
+
+## Coding guidelines
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:47]: Change the heading to - Coding Guidelines

File: developer_guide.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+### Setup development environment
+
+* Go (version 1.17 or later) installed and configured in your machine ([Installing Go](https://go.dev/dl/))
+* Docker installed and running in your local machine
+* A running [`kind`](https://kind.sigs.k8s.io/)  cluster
+* [`kubectl`](https://kubernetes.io/docs/tasks/tools/) installed and configured
+
+
+
+### Build docker images
```
Comment (veenamj-avesha) [inline developer_guide.md:18]: Change the heading to - Building Docker Images

**Review by pnavali — APPROVED** [review]

**Pull request merged**